TiDB crash for concurrently multiple writes are not allowed when lighting/br import

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
lightning/br import million tables
a tidb crash
```
[2025/01/05 07:33:09.177 +00:00] [ERROR] [misc.go:116] ["panic in the recoverable goroutine"] [label=domain] [funcInfo=loadSchemaInLoop] [r="concurrently multiple writes are not allowed"] [stack="github.com/pingcap/tidb/pkg/util.Recover\n\t/workspace/source/tidb/pkg/util/misc.go:120\nruntime.gopanic\n\t/root/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/panic.go:785\ngithub.com/pingcap/tidb/pkg/infoschema.btreeSet[...]\n\t/workspace/source/tidb/pkg/infoschema/infoschema_v2.go:88\ngithub.com/pingcap/tidb/pkg/infoschema.(*Data).add\n\t/workspace/source/tidb/pkg/infoschema/infoschema_v2.go:227\ngithub.com/pingcap/tidb/pkg/infoschema.(*Builder).addTable\n\t/workspace/source/tidb/pkg/infoschema/builder.go:991\ngithub.com/pingcap/tidb/pkg/infoschema.applyCreateTable\n\t/workspace/source/tidb/pkg/infoschema/builder.go:702\ngithub.com/pingcap/tidb/pkg/infoschema.(*Builder).applyTableUpdateV2\n\t/workspace/source/tidb/pkg/infoschema/infoschema_v2.go:1405\ngithub.com/pingcap/tidb/pkg/infoschema.applyTableUpdate\n\t/workspace/source/tidb/pkg/infoschema/infoschema_v2.go:1266\ngithub.com/pingcap/tidb/pkg/infoschema.applyDefaultAction\n\t/workspace/source/tidb/pkg/infoschema/builder.go:320\ngithub.com/pingcap/tidb/pkg/infoschema.(*Builder).ApplyDiff\n\t/workspace/source/tidb/pkg/infoschema/builder.go:105\ngithub.com/pingcap/tidb/pkg/domain.(*Domain).tryLoadSchemaDiffs\n\t/workspace/source/tidb/pkg/domain/domain.go:645\ngithub.com/pingcap/tidb/pkg/domain.(*Domain).loadInfoSchema\n\t/workspace/source/tidb/pkg/domain/domain.go:349\ngithub.com/pingcap/tidb/pkg/domain.(*Domain).Reload\n\t/workspace/source/tidb/pkg/domain/domain.go:786\ngithub.com/pingcap/tidb/pkg/domain.(*Domain).loadSchemaInLoop\n\t/workspace/source/tidb/pkg/domain/domain.go:1149\ngithub.com/pingcap/tidb/pkg/domain.(*Domain).Start.func2\n\t/workspace/source/tidb/pkg/domain/domain.go:1534\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1\n\t/workspace/source/tidb/pkg/util/wait_group_wrapper.go:103"]
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
Release Version: v9.0.0-alpha-48-g76e0fcdfb4
Edition: Community
Git Commit Hash: 76e0fcdfb42b013c397cac4738927e28dc1ecd8c
Git Branch: HEAD
UTC Build Time: 2025-01-05 03:05:58
GoVersion: go1.23.4
Race Enabled: false
Check Table Before Drop: false
Store: unistore

--- a/pkg/infoschema/test/infoschemav2test/v2_test.go
+++ b/pkg/infoschema/test/infoschemav2test/v2_test.go
@@ -18,12 +18,10 @@ import (
 	"context"
 	"fmt"
 	"math"
-	"math/rand"
 	"slices"
 	"sort"
 	"strconv"
 	"strings"
-	"sync"
 	"testing"
 	"time"
 

